python312Packages.textual: 0.72.0 -> 0.79.0

greg-hellings opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

greg-hellings committed:
python312Packages.textual: 0.72.0 -> 0.79.0

pbsds commented on pkgs/development/python-modules/textual/default.nix:
```diff
@@ -39,15 +40,13 @@ buildPythonPackage rec {
-      tree-sitter
-      tree-sitter-languages
-    ];
+    syntax = [ tree-sitter ] ++ lib.optionals (pythonOlder "3.12") [ tree-sitter-languages ];
```
why limit the tree-sitter-languages package?

greg-hellings replied:
Because of [this line](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/tree-sitter-languages/default.nix#L62), which prevents this package from building with versions of python newer than 3.11 if tree-sitter-languages is included as a dep in 3.12 and later.

pbsds replied:
```suggestion
    syntax = [ tree-sitter ] ++ lib.optionals (!tree-sitter-languages.meta.broken) [ tree-sitter-languages ];
```

greg-hellings replied:
Nice. Much cleaner solution.

pbsds commented:
the [build failure](https://logs.ofborg.org/?key=nixos/nixpkgs.333360&attempt_id=2660e8f1-e1b9-4e3b-a151-1b47d073d621) strikes me as a concurrency issue, consider removing pytest-xdist

greg-hellings commented:
> the [build failure](https://logs.ofborg.org/?key=nixos/nixpkgs.333360&attempt_id=2660e8f1-e1b9-4e3b-a151-1b47d073d621) strikes me as a concurrency issue, consider removing pytest-xdist

Correct. I checked with upstream and they suggest adding the flag I put in with the last edit to force group execution. Some of their tests rely on state from previous tests in the same group.

pbsds commented:
but it still breaks

greg-hellings commented:
> but it still breaks

Where? I added the group argument and it now passes, because it runs all the related tests from the group in a single thread. The run you linked appears to be from before I added the extra flag, from what I can tell.

pbsds commented:
![image](https://github.com/user-attachments/assets/d74a0dfd-25c4-4223-b53a-374c62891a2b)
https://github.com/NixOS/nixpkgs/pull/333360/checks?check_run_id=29127535637
https://logs.ofborg.org/?key=nixos/nixpkgs.333360&attempt_id=2660e8f1-e1b9-4e3b-a151-1b47d073d621

greg-hellings commented:
> ![image](https://private-user-images.githubusercontent.com/140964/361157148-d74a0dfd-25c4-4223-b53a-374c62891a2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5NTU0NDksIm5iZiI6MTcyNDk1NTE0OSwicGF0aCI6Ii8xNDA5NjQvMzYxMTU3MTQ4LWQ3NGEwZGZkLTI1YzQtNDIyMy1iNTNhLTM3NGM2Mjg5MWEyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyOVQxODEyMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNmEyMTVhZWU2MTkwMzc2NTQ0YTljNjVjOGYxN2NmZDZjMTA0NTExOTFlODYxMGU0N2ZjY2Q0N2QyMTcyMzQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.J3t0NqdU-QtP5y8nQgucVpFgTsFxNCS8YZcbtoDVkPk) [#333360 (checks)](https://github.com/NixOS/nixpkgs/pull/333360/checks?check_run_id=29127535637) [logs.ofborg.org?key=nixos/nixpkgs.333360&attempt_id=2660e8f1-e1b9-4e3b-a151-1b47d073d621](https://logs.ofborg.org/?key=nixos/nixpkgs.333360&attempt_id=2660e8f1-e1b9-4e3b-a151-1b47d073d621)

That's from the previous commit where I hadn't added the options, from what I can tell.

taha-yassine commented:
Any chance this gets merged soon? I need it for a downstream dependency.

pbsds commented on pkgs/development/python-modules/textual/default.nix:
```diff
@@ -41,13 +44,13 @@ buildPythonPackage rec {
       tree-sitter
-      tree-sitter-languages
-    ];
+    ] ++ lib.optionals (tree-sitter-languages.meta.broken) [ tree-sitter-languages ];
```
```suggestion
    ] ++ lib.optionals (!tree-sitter-languages.meta.broken) [ tree-sitter-languages ];
```

greg-hellings replied:
Good catch
